### PR TITLE
Publish edge tag to local registry as we do on dockerhub

### PR DESCRIPTION
**Pull request by uCatu**
### What does this PR do?
For security, performance & early debugging reasons we want `edge` tag to be present and stored at out local Docker registry as it is at DockerHub.
See [Slack Thread](https://conjurhq.slack.com/archives/C8BNMU0KV/p1617353627026700) for additional details

### What ticket does this PR close?
N/A

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -534,7 +534,15 @@ pipeline {
...
+            sh './push-image.sh --edge --registry-prefix=registry.tld'
+          }
+        }
+        
+        stage('On a master build') {
+          when { branch "master" }
+          // Parallel publish edge tag, latest green main builds, to DockerHub
           steps {
             sh './push-image.sh --edge'
           }
```
Comment (orenbm) [inline Jenkinsfile:548]: this seems odd. can we have an additional stage inside `On a master build` that will run the tasks in parallel instead? something like this: https://github.com/cyberark/secrets-provider-for-k8s/blob/master/Jenkinsfile#L113

Comment (uCatu) [reply Jenkinsfile:548]: @orenbm As I follow existing parallel stage approach in this Jenkisfile I suggest to raise the option for converting it to `tasks` in a follow-up PR that will align all Jenkisfiles if chosen to be our best practice guideline, WDYT?

Comment (orenbm) [reply Jenkinsfile:548]: yes but in this case you have 2 things that you want to do in the same scenario - on a master build.
I would at least change the stage names so they are not the same.

Comment (orenbm) [reply Jenkinsfile:548]: i still think that we should run these tasks in parallel inside the stage. you can also you the current mechanism to do that by adding an additional `parallel` inside the stage.

File: Jenkinsfile [hunk 2]
```diff
@@ -534,9 +534,18 @@ pipeline {
...
+             script {
+               def tasks = [:]
+               tasks["Publish edge to local registry"] = {
+                 sh './push-image.sh --edge --registry-prefix=registry.tld'
+               }
+               tasks["Publish edge to DockerHub"] = {
+                 sh './push-image.sh --edge'
+               }
+               parallel tasks
+             }
```
Comment (orenbm) [inline Jenkinsfile:548]: please indent this one space backwards

File: Jenkinsfile [hunk 3]
```diff
@@ -534,9 +534,18 @@ pipeline {
 
         stage('On a master build') {
           when { branch "master" }
-
+          // Parallel publish edge tag image
```
Comment (orenbm) [inline Jenkinsfile:537]: I think
```suggestion
          // Parallel publish edge tag image
```
I think that this comment is redundant but if you do write it please change it to
```
// Publish edge tag image to private and public registries in parallel
```

**Review by orenbm — CHANGES_REQUESTED** [review]
left some comments

**Review by orenbm — APPROVED** [review]

**Pull request merged**